lucille-core/src/main/java/com/kmwllc/lucille/util/OpenSearchUtils.java

```diff
@@ -1,73 +1,110 @@
 package com.kmwllc.lucille.util;
 
 import com.typesafe.config.Config;
-import nl.altindag.ssl.SSLFactory;
-import org.apache.http.HttpHost;
-import org.apache.http.auth.AuthScope;
-import org.apache.http.auth.UsernamePasswordCredentials;
-import org.apache.http.client.CredentialsProvider;
-import org.apache.http.impl.client.BasicCredentialsProvider;
-import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
-import org.opensearch.client.RestClient;
-import org.opensearch.client.RestClientBuilder;
-import org.opensearch.client.RestHighLevelClient;
-
 import java.net.URI;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.SSLContext;
+import org.apache.hc.client5.http.auth.AuthScope;
+import org.apache.hc.client5.http.auth.UsernamePasswordCredentials;
+import org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManagerBuilder;
+import org.apache.hc.client5.http.ssl.ClientTlsStrategyBuilder;
+import org.apache.hc.client5.http.ssl.DefaultHostnameVerifier;
+import org.apache.hc.client5.http.ssl.NoopHostnameVerifier;
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.hc.core5.http.nio.ssl.TlsStrategy;
+import org.apache.hc.core5.ssl.SSLContextBuilder;
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.httpclient5.ApacheHttpClient5TransportBuilder;
 
 /**
  * Utility methods for communicating with OpenSearch.
  */
 public class OpenSearchUtils {
 
+  public static OpenSearchClient getOpenSearchRestClient(Config config) {
+    return getOpenSearchRestClientInternal(config);
+  }
+
   /**
    * Generate a RestHighLevelClient from the given config file. Supports Http OpenSearchClients.
    *
    * @param config The configuration file to generate a client from
    * @return the RestHighLevelClient client
    */
-  public static RestHighLevelClient getOpenSearchRestClient(Config config) {
-
-    // get host uri
-    URI hostUri = URI.create(getOpenSearchUrl(config));
-
-    //Establish credentials to use basic authentication.
-    final CredentialsProvider provider = new BasicCredentialsProvider();
-
-    // get user info from URI if present and setup BasicAuth credentials if needed
-    String userInfo = hostUri.getUserInfo();
-    if (userInfo != null) {
-      int pos = userInfo.indexOf(":");
-      String username = userInfo.substring(0, pos);
-      String password = userInfo.substring(pos + 1);
-      provider.setCredentials(AuthScope.ANY,
-          new UsernamePasswordCredentials(username, password));
-    }
+  public static OpenSearchClient getOpenSearchRestClientInternal(Config config) {
```

Comment:
please add a comment with a link to the place where we adapted the SSL setup from (the example repo you found)
please mention anything that's different about our code from the example provided for opensearch

Reply:
Done